.github/actions/fetch-artifact/action.yml

```diff
@@ -0,0 +1,63 @@
+name: Fetch artifact
+description: Fetches a specific artifact and downloads it to the current directory
+
+inputs:
+  name:
+    description: The name of the artifact to fetch
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: find_release_artifact
+      id: find_release_artifact
+      uses: actions/github-script@v7
+      with:
+        result-encoding: string
+        script: | # js
+          const fs = require('fs');
+
+          const artifacts = await github.rest.actions.listArtifactsForRepo({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            name: "${{ inputs.name }}",
+            per_page: 1,
+          });
+
+          if (!artifacts.data?.artifacts?.[0]?.id) {
+            throw new Error(`No artifacts found for ${{ inputs.commit }}`);
+          }
+
+          const artifact_id = artifacts.data.artifacts[0].id;
+
+          const download = await github.rest.actions.downloadArtifact({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            artifact_id: artifact_id,
+            archive_format: 'zip',
+          });
+
+          fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/mb.zip`, Buffer.from(download.data));
+
+    - name: unzip uberjar artifact
+      shell: bash
+      run: unzip mb.zip
+    - name: Move the Uberjar
+      shell: sh
```

Comment:
Is `sh` allowed? Or did you mean `bash`?

Reply:
yep! `sh` is the default shell, probably no different than bash. I should just use bash for consistency's sake